### ISSUE TITLE
AttributeError: 'module' object has no attribute 'dpi'

### ISSUE THREAD

**Issue by earino** [plotly/plotly.py#179]
Attempting to write a blog post in which I run plotly on domino data labs python notebooks, I am running into this error:

AttributeError: 'module' object has no attribute 'dpi'

The full error trace is available at the domino run url: https://app.dominodatalab.com/earino/domino_plotly#r/54bffa7fe4b06f03e78c6d67

I have made the project public, so feel free to run it.

Full stack trace:

Traceback (most recent call last):
  File "decision.py", line 106, in <module>
    plot_url = py.plot_mpl(plt)
  File "/usr/local/lib/python2.7/dist-packages/plotly/plotly/plotly.py", line 283, in plot_mpl
    fig = tools.mpl_to_plotly(fig, resize=resize, strip_style=strip_style)
  File "/usr/local/lib/python2.7/dist-packages/plotly/tools.py", line 371, in mpl_to_plotly
    matplotlylib.Exporter(renderer).run(fig)
  File "/usr/local/lib/python2.7/dist-packages/plotly/matplotlylib/mplexporter/exporter.py", line 45, in run
    fig.savefig(io.BytesIO(), format='png', dpi=fig.dpi)
AttributeError: 'module' object has no attribute 'dpi'

**Comment by theengineear** [comment]
@earino , are you using matplotlib 1.3.1 or a newer version? We still only support 1.3.1 but intend to support future matplotlib versions.

**Comment by earino** [comment]
Greetings!

Yes, I believe so, if you go here:

https://app.dominodatalab.com/earino/domino_plotly

requirements.txt has:
matplotlib==1.3.1

(url: https://app.dominodatalab.com/earino/domino_plotly/view/requirements.txt?commitId=55e764a847fd3cdbd98f7f40435eb5aa5d2d85e9)

and the code is:
https://app.dominodatalab.com/earino/domino_plotly/view/decision.py?commitId=55e764a847fd3cdbd98f7f40435eb5aa5d2d85e9

basically, i am trying to convert the canonical example of classifier decision boundaries into a nice plotly graph.

What other debug information can i provide?

**Comment by theengineear** [comment]
@earino 

Looks like you passed `plt` into the py.plot_mpl() function. Try changing the following:

```python
figure.subplots_adjust(left=.02, right=.98)
#plt.show()
py.sign_in(os.environ['PLOTLY_USER'], os.environ['PLOTLY_PASSWORD'])
plot_url = py.plot_mpl(plt)
```

To:

```python
figure.subplots_adjust(left=.02, right=.98)
#plt.show()
py.sign_in(os.environ['PLOTLY_USER'], os.environ['PLOTLY_PASSWORD'])
plot_url = py.plot_mpl(figure)  # <-- right here
```

Let us know if that works! Without an idea of what's getting visualized, I can't predict if it will copy over. However, if it doesn't work, I'd happily show you a way to plot with the plotly module alone.

**Comment by earino** [comment]
Well, I'm embarrased. That worked, in that it generated a plot.

Now, however, the problem is that the plot is fundamentally different :) So, the original plot I'm trying to produce is from here: http://scikit-learn.org/stable/auto_examples/plot_classifier_comparison.html

it is the best visualization i have ever seen of "classifier decision boundaries." I literally took the code from this example, and added a call to py.plot_mpl() and it gave me:

https://plot.ly/~earinodomino/10

Which is close, but definitely very different. In the run log I see the warning message:

     Dang! That path collection is out of this world. I totally don't know what to do with it yet! Plotly can only   import path collections linked to 'data' coordinates

You can see the whole run-log here: https://app.dominodatalab.com/earino/domino_plotly/results/54c27befe4b0f38272ad8711

Is this simply a limitation of plotly, is there anything I can do differently?

**Comment by theengineear** [comment]
@earino , the major issue is that matplotlib stores images for those heatmaps and we don't attempt to infer when images are data and what to do with it.

However, we also thought this plot was great and used it as an example which you can find here:

http://nbviewer.ipython.org/github/etpinard/plotly-misc-nbs/blob/etienne/ml-classifier-comp/ml-classifier-comp.ipynb

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
